Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,240 @@
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+WI.MediaDetailsSidebarPanel = class MediaDetailsSidebarPanel extends WI.DOMDetailsSidebarPanel
+{
+    constructor(delegate)
+    {
+        super("media-details", WI.UIString("Media"));
+        this.element.classList.add("media-details");
+    }
+
+    // DOMDetailsSidebarPanel Overrides
+
+    supportsDOMNode(nodeToInspect)
+    {
+        return nodeToInspect.isMediaElement();
+    }
+
+    // Protected
+
+    initialLayout()
+    {
+        super.initialLayout();
+
+        this._sourceRow = new WI.DetailsSectionSimpleRow(WI.UIString("Source"));
+        this._viewportRow = new WI.DetailsSectionSimpleRow(WI.UIString("Viewport"));
+        this._framesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Frames"));
+        this._resolutionRow = new WI.DetailsSectionSimpleRow(WI.UIString("Resolution"));
+        this._videoFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Format"));
+        this._audioFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Format"));
+
+        let generalGroup = new WI.DetailsSectionGroup([this._sourceRow, this._viewportRow, this._framesRow, this._resolutionRow, this._videoFormatRow, this._audioFormatRow
+                                                       ]);
+        this._generalSection = new WI.DetailsSection("media-details-general", WI.UIString("General"), [generalGroup]);
+        this.element.appendChild(this._generalSection.element);
+
+        this._videoCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Codec"));
+        this._transferRow = new WI.DetailsSectionSimpleRow(WI.UIString("Transfer Function"));
+        this._primariesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Primaries"));
+        this._matrixRow = new WI.DetailsSectionSimpleRow(WI.UIString("Matrix Coefficients"));
+        this._fullRangeRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Range"));
+
+        let videoGroup = new WI.DetailsSectionGroup([this._videoCodecRow, this._primariesRow, this._transferRow, this._matrixRow, this._fullRangeRow]);
+        this._videoSection = new WI.DetailsSection("media-video-details", WI.UIString("Video Details"), [videoGroup]);
+        this.element.appendChild(this._videoSection.element);
+
+        this._audioCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Codec"));
+        this._sampleRateRow = new WI.DetailsSectionSimpleRow(WI.UIString("Sample Rate"));
+        this._channelsRow = new WI.DetailsSectionSimpleRow(WI.UIString("Channels"));
+
+        let audioGroup = new WI.DetailsSectionGroup([this._audioCodecRow, this._sampleRateRow, this._channelsRow]);
+        this._audioSection = new WI.DetailsSection("media-audio-details", WI.UIString("Audio Details"), [audioGroup]);
+        this.element.appendChild(this._audioSection.element);
+    }
+
+    async layout()
```

Comment:
Making `layout` async is likely to be unpredictable, since this means that children views (which admittedly you don't have any here) would get laid out before this view has finished laying out.

Generally layout is not when we should be doing the work of getting the underlying data from the backend. This could be particularly costly during resize, as `layout` will be called once per frame during a resize.

One possible solution would be to perform the asynchronous work in a private function you invoke with your update timer and then call `needsLayout` once you have updated info available.

(Also keep in mind that calls to the backend are not instant, particularly when inspecting content on another device via Remote Inspection.)

Reply:
Frankly, `needsLayout()` and `layout()` seem to be entirely mis-used in a lot of the Web Inspector code, since the implementations are about updating UI from the model, and not about layout itself. But sure, I'll refactor all this to move the pushing of new values into the UI outside `layout()`.

Reply:
To be clear, my concern is that we shouldn’t be doing async work inside layout to ask the frontend’s model to be updated from the backend, I don’t see an issue with updating from the frontend’s current representation of the model here.

Reply:
Right, but updating the DOM from the model isn't _layout_. Perhaps updating the DOM from the model should _trigger layout_, but it isn't layout in and of itself.

Reply:
Regardless, the async updating of information was moved out of the `layout()` method (which no longer exists because it's not needed; CSS is sufficient here).